lib/tasks/metrics.rake

```diff
@@ -0,0 +1,30 @@
+namespace :metrics do
+  desc 'Exports metrics in all the popular formats to s3'
+  task export: :environment do
+    abort 'Please set the env-var S3_METRICS_BUCKET_NAME' if ENV['S3_METRICS_BUCKET_NAME'].blank?
+
+    # get list of all available metrics
+    METRIC_CLASSES = [Metrics::Moves].map { |namespace|
+      namespace.constants.map { |c| namespace.const_get(c) if namespace.const_get(c).is_a? Class }.compact
+    }.flatten
+
+    s3 = Aws::S3::Resource.new
```

Comment:
Instead of defining out own s3 bucket file writer, we can utilise the following class: https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/lib/cloud_data_feed.rb which is doing the same thing (except for the naming)

Reply:
spoke in person: the references class doesn't support what we need here as the file structure is different and this is used heavily in feeds. It might be safer to just create our own class with tests to simplify the logic in the rake task